File: test-suite-mcp-http-tck-common/src/main/java/io/modelcontextprotocol/server/http/tck/async/Prompts.java
package io.modelcontextprotocol.server.http.tck.async;

import org.jspecify.annotations.NonNull;
import io.micronaut.mcp.annotations.Prompt;
import io.micronaut.mcp.annotations.PromptArg;
import io.modelcontextprotocol.spec.McpSchema;
import jakarta.inject.Singleton;

import java.util.List;

import static io.modelcontextprotocol.spec.McpSchema.Role.USER;

@Singleton
class Prompts {
    @Prompt(
        name = "code_review",
        title = "Request Code Review",
        description = "Asks the LLM to analyze code quality and suggest improvements"
    )
    McpSchema.GetPromptResult codeReview(@PromptArg(description = "The code to review") @NonNull String code) {
        return new McpSchema.GetPromptResult("Code review prompt",
            List.of(new McpSchema.PromptMessage(USER, new McpSchema.TextContent("Please review this Python code"))));
    }
}

File: test-suite-docs-python/src/test/python/micronaut/mcp/docs/prompts/Prompts.py
from micronaut.context.annotation import Requires
# tag::imports[]
from typing import Annotated

from jakarta.inject import Singleton
from micronaut.mcp.annotations import Prompt, PromptArg
# end::imports[]


@Requires(property="spec.name", value="PromptsTest")
# tag::clazz[]
@Singleton
class Prompts:
    @Prompt(name="chess-statistics", description="Displays statistics for chess games")
    def prompt(self, name: Annotated[str, PromptArg(description="Player Name")]) -> str:
        """
        :return: Chess statistics
        """
        return f"You generate chess statistics for {name} ...."
# end::clazz[]
